node_modules/col-course/lib/api.js

```diff
@@ -31,22 +31,30 @@ var log = require('col-core/lib/logger')('col-course');
 /**
  * Get a course. If the course doesn't exist yet, it will be created
  *
- * @param  {Number}       canvasCourseId      The id of the course in Canvas
- * @param  {Canvas}       canvas              The Canvas instance the course is running on
- * @param  {Function}     callback            Standard callback function
- * @param  {Object}       callback.err        An error object, if any
- * @param  {Course}       callback.course     The retrieved or created course
+ * @param  {Number}       canvasCourseId                    The id of the course in Canvas
+ * @param  {Canvas}       canvas                            The Canvas instance the course is running on
+ * @param  {Object}       courseInfo                        Additional info for the course
+ * @param  {String}       [courseInfo.assetlibrary_url]     The URL where the asset library in this course can be reached
+ * @param  {String}       [courseInfo.whiteboards_url]      The URL where the whiteboards in this course can be reached
+ * @param  {Function}     callback                          Standard callback function
+ * @param  {Object}       callback.err                      An error object, if any
+ * @param  {Course}       callback.course                   The retrieved or created course
  */
-var getOrCreateCourse = module.exports.getOrCreateCourse = function(canvasCourseId, canvas, callback) {
+var getOrCreateCourse = module.exports.getOrCreateCourse = function(canvasCourseId, canvas, courseInfo, callback) {
   // Parameter validation
   var validationSchema = Joi.object().keys({
     'canvasCourseId': Joi.number().required(),
-    'canvas': Joi.object().required()
+    'canvas': Joi.object().required(),
+    'courseInfo': Joi.object().keys({
+      'assetlibrary_url': Joi.string().optional(),
+      'whiteboards_url': Joi.string().optional()
+    }).required()
```

Comment:
Will there ever be a time where a course is created without an asset library / whiteboard url?